frontend/src/concepts/roleBinding/RoleBindingPermissionsTable.tsx

```diff
@@ -52,7 +51,53 @@ const RoleBindingPermissionsTable: React.FC<RoleBindingPermissionsTableProps> =
   refresh,
 }) => {
   const [editCell, setEditCell] = React.useState<string[]>([]);
-
+  const createProjectRoleBinding = (
+    subjectName: string,
+    newRBObject: RoleBindingKind,
+    rb: RoleBindingKind,
+  ) => {
+    const roleType = newRBObject.subjects[0].kind.toLowerCase();
+    const usedNames: string[] = permissions
+      .map((p) => p.subjects[0].name)
+      .filter((name) => isAdding || name !== rb.subjects[0].name);
+    const isDuplicateName = usedNames.includes(subjectName);
+    if (isDuplicateName) {
+      onError(
+        <>
+          The {roleType} <strong>{subjectName}</strong> already exists. Edit the {roleType}
+          &apos;s existing permissions, or add a new {roleType} to assign it permissions.
+        </>,
+      );
+      refresh();
+    } else if (isAdding) {
+      createRoleBinding(newRBObject)
+        .then(() => {
+          onDismissNewRow();
+          refresh();
+        })
+        .catch((e) => {
+          onError(<>{e}</>);
+        });
+    } else {
+      createRoleBinding(newRBObject)
+        .then(() =>
+          deleteRoleBinding(rb.metadata.name, rb.metadata.namespace)
+            .then(() => refresh())
+            .catch((e) => {
+              onError(<>{e}</>);
+              setEditCell((prev) => prev.filter((cell) => cell !== rb.metadata.name));
```

Comment:
Could you give some explanation to this section? If not a duplicate and not isAdding then we create a new RB and then delete the old one? What case is this covering? on edit?